src/main/java/heronarts/lx/midi/surface/MidiFighterTwister.java

```diff
@@ -337,12 +350,57 @@ public void onParameterChanged(LXParameter parameter) {
       }
     }
 
+    private boolean isKnobRelative(int index) {
+      // Fixed for now but could be expanded to poll MFT encoder settings on startup with sysex
```

Comment:
Is it possible to set the knob mode via MIDI? Or does the MFT user need to know that they need to do some button presses on the MFT to get this configured properly?

Ideally the control surface mode should activate the appropriate mode on all the encoders.

Reply:
No the knob mode can't be set via MIDI or by button presses.  You have to do it through the Midi Fighter Utility application.  It is a downside, anyone running a MFT needs to configure it for their application... in this case apply the LX-specific settings.  I have notes about the proper settings at the top of the MFT file.

Well... that's not entirely true.  The Midi Fighter Utility writes all the system settings and encoder settings to the device with a long (or maybe multiple?) sysex message.  (system settings + 64 virtual knobs*[11 settings per knob] is a fairly long list.)  So we *could* mimic the device-flash sysex and set all the settings to be LX-compatible on startup.  That seems a bit bold... but OTOH it sure would eliminate any issues with people failing to get the settings right and having issues running them with LX.

Doing that would overwrite a user's previous settings on the MFT... so if they had a custom configuration for some other software they would need to export their settings through the Utility software and re-flash them when switching back.  I guess they would have to do that anyway.  We could try to play nice by reading the device settings on initialization, sending LX settings, and then re-send original settings before shutdown.  No avoiding losing the settings though on a non-clean shutdown.  But hey, it would be slick for people in a stable environment.

Reply:
> We could try to play nice by reading the device settings on initialization, sending LX settings, and then re-send original settings before shutdown. No avoiding losing the settings though on a non-clean shutdown. But hey, it would be slick for people in a stable environment.

If this is possible that would be great. It seems analogous to me to how Ableton puts the APC40 into Live Mode when it's activated as a control surface, and then puts it back afterwards when it's done. If we were able to do the same thing with the MFT, that'd be the most slick approach and would ensure that it's always behaving properly in control surface mode.

Reply:
Looks like this is achievable.  If it's okay with you can we leave this for a later pull request?  Might not get done before Treefort and could be a mini-project of its own.